google/cloud/storage/parallel_upload.h

```diff
@@ -750,9 +748,9 @@ Composer ResumableParallelUploadState::CreateComposer(
   auto get_metadata_options = StaticTupleFilter<
       Among<DestinationPredefinedAcl, EncryptionKey, KmsKeyName, QuotaUser,
             UserIp, UserProject, WithObjectMetadata>::TPred>(options);
-  auto composer =
-      [client, bucket_name, object_name, compose_options, get_metadata_options,
-       prefix](std::vector<ComposeSourceObject> const& sources) mutable
+  auto composer = [client, bucket_name, object_name, compose_options,
+                   get_metadata_options,
+                   prefix](std::vector<ComposeSourceObject> sources) mutable
```

Comment:
The `Composer` alias is defined as

https://github.com/googleapis/google-cloud-cpp/blob/7cb938b386da73a3adc088646c19aa18bd6c581f/google/cloud/storage/parallel_upload.h#L124-L127

So will changing this signature here to accept by value actually eliminate a copy somewhere?

Reply:
That is either terrible or hilarious... see the lambda does get assigned to a `Composer` but that goes to:


https://github.com/googleapis/google-cloud-cpp/blob/7cb938b386da73a3adc088646c19aa18bd6c581f/google/cloud/storage/parallel_upload.h#L226-L227

which has a different signature, but for extra fun it was used here:

https://github.com/googleapis/google-cloud-cpp/blob/7cb938b386da73a3adc088646c19aa18bd6c581f/google/cloud/storage/parallel_upload.cc#L253

I think all this stuff needs some deep cleaning.